### ISSUE TITLE
Design of activation calculations

### ISSUE THREAD

**Issue by paulromano** [openmc-dev/openmc#1611]
Right now `openmc.deplete` only works for reactor depletion calculations. I am working with @Shimwell to extend `openmc.deplete` to perform activation calculations (i.e., using a fixed source) and wanted to get some opinions on the design, mostly from those who have been involved in depletion efforts (@drewejohnson @liangjg @cjosey). A few of the design considerations include:

1. Our `Integrator` classes take a `power` or `power_density` argument. For activation calculations, we'll need to specify the source rate. Technically, we could rely on `Source.strength` and not supply it at all to `Integrator`, but it seems like it would be prudent to allow the strength to be changed just like we do with `power`. Also, if you want to have some steps be decay-only, specifying a strength of zero seems like the most straightforward way.
2. That `power` argument is passed through a series of calls: `Integrator.__call__`, `Operator.__call__`, `Operator._unpack_tallies_and_normalize`, etc. For the fixed source mode, what we need to pass is the absolute source rate in neutrons/sec, not a power. I'm not sure what a good name is for a variable that could be a power or an external source rate, but I'm considering `source_rate`. I thought about just `rate` (since power is an energy rate after all), but we use "rate" a lot to refer to reaction rates, so I feel like that will get confusing. Thoughts?
3. Our energy helper classes are specific to normalizing by energy deposition. For fixed source, we will just normalize by the source rate. I'm planning on renaming `EnergyHelper` -> `NormalizationHelper` and adding another subclass for the fixed source case. One design question is whether it should be the user's responsibility to specify the helper. Between the different argument in `Integrator` and the fact that the `Settings` object under `Operator` will indicate a fixed source run mode, it seems like we can automatically select the helper.

**Comment by drewejohnson** [comment]
This sounds wonderful. I will put my responses in the same enumeration, but generally I think this is a good approach

1. Specifying a `source_rate` the same way we provide power and power density seems like a consistent design. I've got some thoughts on the decay calculation, that I'll put at the end as this could also touch reactor depletion studies
2. For the passing of ``power``, would a name like ``normalization_rate`` be too verbose? It's definitely the most general, but almost feels too ambiguous
3. Unless I'm missing something, I think we can determine the use of a "source-based" normalization helper from the inputs as you indicated. However, we already have the `energy_mode` argument in the `Operator` that switches between energy deposition and normalizing based on fission q values. If we do rely on other inputs to determine the source-based normalization, is that the "correct" way to determine normalization? I would lean towards having another option in the `Operator.__init__`, but maybe checking against the `Settings` passed? The `Integrator` also doesn't know about the `Operator`s normalization mode, nor does the `Operator` know about the meaning behind the power or normalization quantity provided by the `Integrator`

With respect to the decay mode, setting a normalization target of zero will result in the zero-valued reaction rates necessary to build a pure-decay depletion matrix. The downside is a full transport solution would still be performed at BOS to compute those unused reaction rates. Possible implementations for creating / emulating zero reaction rates w/o running a full transport solve

1. Make a list of zero-valued `reaction_rates` to be passed to [`Chain.form_matrix`](https://docs.openmc.org/en/latest/pythonapi/generated/openmc.deplete.Chain.html#openmc.deplete.Chain.form_matrix) during the decay solver. 
2. Exploit some duck-typing and make a class that looks like an array of reaction rates but always returns zero upon indexing
3. Add a dedicated `Chain.build_decay_matrix` method that doesn't accept reaction rates. Since this is only dependent on nuclear data, we could store it for later steps, assuming all materials track the same nuclides which I'm pretty sure they do. Then we don't have to re-build it at every step for all materials.

**Comment by paulromano** [comment]
Ok, so it sounds like we agree on 1 from the original list. For 2, I do think `normalization_rate` is a little verbose/ambiguous for my liking. Another possibility is an explicit name like `power_or_source`, but I think I would still lean toward `source_rate` in the absence of better alternatives. For 3, I realized after I wrote the original message about the disconnect between `Operator` and `Integrator`. I think in my first shot at an implementation, I'll probably go the simplest route and just have another normalization mode that is explicitly passed (like you said, perhaps checking against the settings to make sure it makes sense).

You're absolutely right about the decay mode. I would add one other note which is that we always perform a transport calculation at the last time point, which is not desirable in an activation calculation. My planned approach here is to first implement the capability for doing fixed source activation calculations and then once that is in, address the inefficiencies related to extra transport steps. Also, nice idea with respect to building the decay matrix once!
